pkg/controller/registry/registry_controller.go

```diff
@@ -98,36 +96,21 @@ func (r *ReconcileRegistry) Reconcile(request reconcile.Request) (reconcile.Resu
 	originalRegistry := registry.DeepCopy()
 
 	// Fetch the Instance
-	harbor := &registriesv1alpha1.Instance{}
-	ns := types.NamespacedName{
-		Namespace: registry.Namespace,
-		Name:      registry.Spec.ParentInstance.Name,
-	}
-	err = r.client.Get(ctx, ns, harbor)
-	if errors.IsNotFound(err) {
-		registry.Status = registriesv1alpha1.RegistryStatus{
-			Name:           string(registriesv1alpha1.RegistryStatusPhaseCreating),
-			Message:        "corresponding harbor instance does not exist",
-			LastTransition: &now,
+	harbor, err := internal.FetchReadyHarborInstance(ctx, registry.Namespace, registry.Spec.ParentInstance.Name, r.client)
+	if err != nil {
+		if err == internal.ErrInstanceNotFound {
+			registry.Status = registriesv1alpha1.RegistryStatus{Name: string(registriesv1alpha1.RegistryStatusPhaseCreating)}
+			// Requeue, the instance might not have been created yet
+			return reconcile.Result{RequeueAfter: time.Second * 30}, nil
```

Comment:
For consistency I would format the value of `RequeueAfter` the same way both times:

```suggestion
			return reconcile.Result{RequeueAfter: 30 * time.Second}, nil
```